### DISCUSSION TITLE
Cannot access data in Plotly Panel

### DISCUSSION THREAD

**Discussion by HenrikRolandHansen** [grafana/grafana-csv-datasource#76] [Q&A] [answered]
First Grate plugin - really a missing part of the Grafana toolbox, and now it is here ;)

Not sure if it is a bug or feature request or Plotly panel issue, sorry for that. 

I'm currently trying to visualize data from the CSV plugin via the Plotly Plugin panel (AE3E) but it i'm not able to make it work. 
The CSV plugin returns data in the Table plugin and also other plugins as seen below 
![image](https://user-images.githubusercontent.com/8969659/120307602-77e02b80-c2d3-11eb-9749-9e300536dd6a.png)

The data is normally accessed with "data.series[0].fields[1..x].values.buffer" in the plotly panel plugin, but It does not return any data(so it seams). 

![image](https://user-images.githubusercontent.com/8969659/120308275-38660f00-c2d4-11eb-96fa-65cd8b64fb84.png)

Is there a other way to call the data via the Script?

**Comment by marcusolsson** [comment]
I'm not sure why the Plotly panel isn't returning the data, but I don't know that plugin too well. You can also see the response returned by the data source if you open the Panel Inspector and select **DataFrame structure**.

![Screenshot 2021-06-01 at 14 10 02@2x](https://user-images.githubusercontent.com/8396880/120321026-1cb63500-c2e3-11eb-99b3-2bc344066a76.png)

**Comment by HenrikRolandHansen** [comment, marked as answer]
Thanks for the tip. The returned data is within "_chunks" so the structure is a bit different.


![image](https://user-images.githubusercontent.com/8969659/120328866-8fc3a980-c2eb-11eb-85e8-20cf4dad7d5f.png)

Structure from influxdb query 
![image](https://user-images.githubusercontent.com/8969659/120329190-e29d6100-c2eb-11eb-9e5d-82a77d8e5d32.png)

Anyway was still not able to read the data out.

**Reply by marcusolsson** [reply]
It looks like the JSON Path query returns an object, rather than an array of strings, which would explain why the query works but you're not seeing anything (though it indeed seems to work for the Table panel 🤔 ).

Is there a way you could share a anonymized response so that I could try it out for myself?

**Reply by HenrikRolandHansen** [reply]
Yes it seam to be the problem. Attached a short version of the data(CSV file) and the data response from Grafana.

[data.txt](https://github.com/marcusolsson/grafana-csv-datasource/files/6576890/data.txt)

{
  "state": "Done",
  "series": [
    {
      "fields": [
        {
          "name": "Kabel",
          "type": "string",
          "values": {
            "_nullCount": -1,
            "_type": {},
            "_chunks": [
              [
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable2KFAB",
                "cable2",
                "cable2",
                "cable2",
                "cable3",
                "cable3",
                "cable3",
                "cable3"
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "Kabel",
              "type": {},
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable2KFAB",
              "cable2",
              "cable2",
              "cable2",
              "cable3",
              "cable3",
              "cable3",
              "cable3"
            ]
          },
          "config": {},
          "state": {
            "displayName": null,
            "scopedVars": {
              "__series": {
                "text": "Series",
                "value": {
                  "name": "temp"
                }
              },
              "__field": {
                "text": "Field",
                "value": {}
              }
            },
            "seriesIndex": 1
          }
        },
        {
          "name": "Latitude",
          "type": "number",
          "values": {
            "_nullCount": 0,
            "_type": {
              "precision": 2
            },
            "_chunks": [
              [
                55.17228,
                55.17345,
                55.17282,
                55.17636,
                55.22844,
                55.2724,
                55.28368,
                55.32372,
                55.04882,
                55.05027,
                55.06063,
                55.06228,
                54.9911,
                54.99272,
                55.00071,
                55.00914
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "Latitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              55.17228,
              55.17345,
              55.17282,
              55.17636,
              55.22844,
              55.2724,
              55.28368,
              55.32372,
              55.04882,
              55.05027,
              55.06063,
              55.06228,
              54.9911,
              54.99272,
              55.00071,
              55.00914
            ]
          },
          "config": {},
          "state": {
            "displayName": null,
            "scopedVars": {
              "__series": {
                "text": "Series",
                "value": {
                  "name": "temp"
                }
              },
              "__field": {
                "text": "Field",
                "value": {}
              }
            },
            "calcs": {
              "sum": 882.0188199999998,
              "max": 55.32372,
              "min": 54.9911,
              "logmin": 54.9911,
              "mean": 55.126176249999986,
              "last": 55.00914,
              "first": 55.17228,
              "lastNotNull": 55.00914,
              "firstNotNull": 55.17228,
              "count": 16,
              "nonNullCount": 16,
              "allIsNull": false,
              "allIsZero": false,
              "range": 0.3326199999999986,
              "diff": -0.1631399999999985,
              "delta": 165.39631,
              "step": -0.27490000000000236,
              "diffperc": -0.002956919670530174,
              "previousDeltaUp": true
            },
            "seriesIndex": 2,
            "range": {
              "min": 11.85244,
              "max": 200,
              "delta": 188.14756
            }
          }
        },
        {
          "name": "Longitude",
          "type": "number",
          "values": {
            "_nullCount": 0,
            "_type": {
              "precision": 2
            },
            "_chunks": [
              [
                14.70511,
                14.67364,
                14.6625,
                14.65473,
                14.54748,
                14.45457,
                14.44088,
                14.39617,
                12.92009,
                12.9183,
                12.91033,
                12.90633,
                11.85244,
                11.85482,
                11.85767,
                11.86113
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "Longitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              14.70511,
              14.67364,
              14.6625,
              14.65473,
              14.54748,
              14.45457,
              14.44088,
              14.39617,
              12.92009,
              12.9183,
              12.91033,
              12.90633,
              11.85244,
              11.85482,
              11.85767,
              11.86113
            ]
          },
          "config": {},
          "state": {
            "calcs": {
              "sum": 215.61619,
              "max": 14.70511,
              "min": 11.85244,
              "logmin": 11.85244,
              "mean": 13.476011875,
              "last": 11.86113,
              "first": 14.70511,
              "lastNotNull": 11.86113,
              "firstNotNull": 14.70511,
              "count": 16,
              "nonNullCount": 16,
              "allIsNull": false,
              "allIsZero": false,
              "range": 2.85267,
              "diff": -2.84398,
              "delta": 11.86113,
              "step": -1.4760799999999996,
              "diffperc": -0.19340079740987998,
              "previousDeltaUp": true
            },
            "displayName": null,
            "scopedVars": {
              "__series": {
                "text": "Series",
                "value": {
                  "name": "temp"
                }
              },
              "__field": {
                "text": "Field",
                "value": {}
              }
            },
            "seriesIndex": 3,
            "range": {
              "min": 11.85244,
              "max": 200,
              "delta": 188.14756
            }
          }
        },
        {
          "name": "ProtectionZone",
          "type": "number",
          "values": {
            "_nullCount": 0,
            "_type": {
              "precision": 2
            },
            "_chunks": [
              [
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "ProtectionZone",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200
            ]
          },
          "config": {},
          "state": {
            "calcs": {
              "sum": 3200,
              "max": 200,
              "min": 200,
              "logmin": 200,
              "mean": 200,
              "last": 200,
              "first": 200,
              "lastNotNull": 200,
              "firstNotNull": 200,
              "count": 16,
              "nonNullCount": 16,
              "allIsNull": false,
              "allIsZero": false,
              "range": 0,
              "diff": 0,
              "delta": 0,
              "step": 0,
              "diffperc": 0,
              "previousDeltaUp": true
            },
            "displayName": null,
            "scopedVars": {
              "__series": {
                "text": "Series",
                "value": {
                  "name": "temp"
                }
              },
              "__field": {
                "text": "Field",
                "value": {}
              }
            },
            "seriesIndex": 4,
            "range": {
              "min": 11.85244,
              "max": 200,
              "delta": 188.14756
            }
          }
        },
        {
          "name": "FishingAllowed",
          "type": "boolean",
          "values": {
            "_nullCount": 0,
            "_type": {},
            "_chunks": [
              [
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "FishingAllowed",
              "type": {},
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true
            ]
          },
          "config": {},
          "state": {
            "displayName": null,
            "scopedVars": {
              "__series": {
                "text": "Series",
                "value": {
                  "name": "temp"
                }
              },
              "__field": {
                "text": "Field",
                "value": {}
              }
            },
            "seriesIndex": 5
          }
        }
      ],
      "length": 16,
      "refId": "A",
      "name": "temp",
      "table": {
        "_nullCount": -1,
        "_type": {
          "children": [
            {
              "name": "Kabel",
              "type": {},
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "Latitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "Longitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "ProtectionZone",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "FishingAllowed",
              "type": {},
              "nullable": true,
              "metadata": {}
            }
          ]
        },
        "_chunks": [
          [
            {
              "Kabel": "cable1",
              "Latitude": 55.17228,
              "Longitude": 14.70511,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.17345,
              "Longitude": 14.67364,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.17282,
              "Longitude": 14.6625,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.17636,
              "Longitude": 14.65473,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.22844,
              "Longitude": 14.54748,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.2724,
              "Longitude": 14.45457,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.28368,
              "Longitude": 14.44088,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable1",
              "Latitude": 55.32372,
              "Longitude": 14.39617,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable2KFAB",
              "Latitude": 55.04882,
              "Longitude": 12.92009,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable2",
              "Latitude": 55.05027,
              "Longitude": 12.9183,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable2",
              "Latitude": 55.06063,
              "Longitude": 12.91033,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable2",
              "Latitude": 55.06228,
              "Longitude": 12.90633,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable3",
              "Latitude": 54.9911,
              "Longitude": 11.85244,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable3",
              "Latitude": 54.99272,
              "Longitude": 11.85482,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable3",
              "Latitude": 55.00071,
              "Longitude": 11.85767,
              "ProtectionZone": 200,
              "FishingAllowed": true
            },
            {
              "Kabel": "cable3",
              "Latitude": 55.00914,
              "Longitude": 11.86113,
              "ProtectionZone": 200,
              "FishingAllowed": true
            }
          ]
        ],
        "_chunkOffsets": {
          "0": 0,
          "1": 16
        },
        "_length": 16,
        "_numChildren": 5,
        "_schema": {
          "fields": [
            {
              "name": "Kabel",
              "type": {},
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "Latitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "Longitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "ProtectionZone",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "FishingAllowed",
              "type": {},
              "nullable": true,
              "metadata": {}
            }
          ],
          "metadata": {},
          "dictionaries": {}
        },
        "_children": [
          {
            "_nullCount": -1,
            "_type": {},
            "_chunks": [
              [
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable1",
                "cable2KFAB",
                "cable2",
                "cable2",
                "cable2",
                "cable3",
                "cable3",
                "cable3",
                "cable3"
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "Kabel",
              "type": {},
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable1",
              "cable2KFAB",
              "cable2",
              "cable2",
              "cable2",
              "cable3",
              "cable3",
              "cable3",
              "cable3"
            ]
          },
          {
            "_nullCount": 0,
            "_type": {
              "precision": 2
            },
            "_chunks": [
              [
                55.17228,
                55.17345,
                55.17282,
                55.17636,
                55.22844,
                55.2724,
                55.28368,
                55.32372,
                55.04882,
                55.05027,
                55.06063,
                55.06228,
                54.9911,
                54.99272,
                55.00071,
                55.00914
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "Latitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              55.17228,
              55.17345,
              55.17282,
              55.17636,
              55.22844,
              55.2724,
              55.28368,
              55.32372,
              55.04882,
              55.05027,
              55.06063,
              55.06228,
              54.9911,
              54.99272,
              55.00071,
              55.00914
            ]
          },
          {
            "_nullCount": 0,
            "_type": {
              "precision": 2
            },
            "_chunks": [
              [
                14.70511,
                14.67364,
                14.6625,
                14.65473,
                14.54748,
                14.45457,
                14.44088,
                14.39617,
                12.92009,
                12.9183,
                12.91033,
                12.90633,
                11.85244,
                11.85482,
                11.85767,
                11.86113
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "Longitude",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              14.70511,
              14.67364,
              14.6625,
              14.65473,
              14.54748,
              14.45457,
              14.44088,
              14.39617,
              12.92009,
              12.9183,
              12.91033,
              12.90633,
              11.85244,
              11.85482,
              11.85767,
              11.86113
            ]
          },
          {
            "_nullCount": 0,
            "_type": {
              "precision": 2
            },
            "_chunks": [
              [
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200,
                200
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "ProtectionZone",
              "type": {
                "precision": 2
              },
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200,
              200
            ]
          },
          {
            "_nullCount": 0,
            "_type": {},
            "_chunks": [
              [
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true,
                true
              ]
            ],
            "_chunkOffsets": {
              "0": 0,
              "1": 16
            },
            "_length": 16,
            "_numChildren": 0,
            "_field": {
              "name": "FishingAllowed",
              "type": {},
              "nullable": true,
              "metadata": {}
            },
            "_chunk": [
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true,
              true
            ]
          }
        ]
      }
    }
  ],
  "annotations": [],
  "request": {
    "app": "dashboard",
    "requestId": "Q109",
    "timezone": "browser",
    "panelId": 23763571993,
    "dashboardId": 96,
    "range": {
      "from": "2020-06-01T14:22:58.999Z",
      "to": "2021-06-01T14:22:59.000Z",
      "raw": {
        "from": "now-1y",
        "to": "now"
      }
    },
    "timeInfo": "",
    "interval": "1y",
    "intervalMs": 31536000000,
    "targets": [
      {
        "decimalSeparator": ".",
        "delimiter": ";",
        "experimental": {
          "regex": false
        },
        "header": true,
        "ignoreUnknown": false,
        "path": "Kabelpositionsdata - copy.csv",
        "queryType": "randomWalk",
        "refId": "A",
        "schema": [
          {
            "name": "Kabel",
            "type": "string"
          },
          {
            "name": "Latitude",
            "type": "number"
          },
          {
            "name": "Longitude",
            "type": "number"
          },
          {
            "name": "ProtectionZone",
            "type": "number"
          },
          {
            "name": "FishingAllowed",
            "type": "boolean"
          }
        ],
        "skipRows": 0,
        "datasource": "CSV"
      }
    ],
    "maxDataPoints": 1,
    "scopedVars": {
      "__interval": {
        "text": "1y",
        "value": "1y"
      },
      "__interval_ms": {
        "text": "31536000000",
        "value": 31536000000
      }
    },
    "startTime": 1622557379000,
    "rangeRaw": {
      "from": "now-1y",
      "to": "now"
    },
    "endTime": 1622557379055
  },
  "timeRange": {
    "from": "2020-06-01T14:22:58.999Z",
    "to": "2021-06-01T14:22:59.000Z",
    "raw": {
      "from": "now-1y",
      "to": "now"
    }
  },
  "timings": {
    "dataProcessingTime": 0.014999997802078724
  }
}

**Reply by HenrikRolandHansen** [reply]
Found a solution i Think. It seams that the data is returned as float64array and that Plotly Plugin Panel is not able to use the data in this format. I managed to convert from float64array to array by doing this and now it works.

[].slice.call(data.series[0].fields[1].values._chunk.values)
